tests/connect/standalone-x509-extract_username-001.phpt

```diff
@@ -0,0 +1,37 @@
+--TEST--
+Connect to MongoDB with SSL and X509 auth and username retrieved from cert
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; NEEDS("STANDALONE_X509"); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+$SSL_DIR = realpath(__DIR__ . '/../../scripts/ssl/');
+
+$driverOptions = [
+    // libmongoc does not allow the hostname to be overridden as "server"
+    'allow_invalid_hostname' => true,
+    'weak_cert_validation' => false,
+    'ca_file' => $SSL_DIR . '/ca.pem',
+    'pem_file' => $SSL_DIR . '/client.pem',
+    // TODO: this doesn't appear to have any effect. Does the PEM file not have a password?
```

Comment:
Secure Channel does not support password protected PEM files.
Secure Transport, OpenSSL, LibreSSL and libtls all do.

The password is not needed to extract the subject name from a certificate though.

None of the certificates in the phongo test suite use the password protected private keys, so you don't have to provide one for client authentication. Ignoring password for unencrypted private keys is implementation dependent anyway, I don't know if all of them do, so I wouldn't provide bogus password